### ISSUE TITLE
Support for hasher state dump and recovery

### ISSUE THREAD

**Issue by sticnarf** [nickbabcock/highway-rs#88]
I have a stream of data that may be split into parts and distributed across different nodes. I want to confirm the hash of the stream after splitting. 

Then, a resonable solution is to serialize the hasher state and send it to the node processing the next part after hashing each part.

https://github.com/shepmaster/twox-hash/pull/18 adds serde support to twox-hash. But I doubt whether serde is a good idea because the state variable names are not useful. We can specify the order of state variables, then compact bytes of state variables will be enough.

**Comment by nickbabcock** [comment]
I agree, having a resumable API with opaque serialized state is desirable. Did you have something in mind?
